client/state/data-layer/wpcom/sites/gutenberg/index.js

```diff
@@ -0,0 +1,47 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { EDITOR_TYPE_SET } from 'state/action-types';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { setSiteEditor } from 'state/sites/editor/actions';
+import { bypassDataLayer } from 'state/data-layer/utils';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+
+export const setType = action =>
+	http(
+		{
+			path: `/sites/${ action.siteId }/gutenberg`,
+			method: 'POST',
+			apiNamespace: 'wpcom/v2',
+			query: {
+				editor: action.editor,
+				http_envelope: 1,
+			},
+			body: {},
+		},
+		action
+	);
+
+export const receiveEditorTypeError = action => {
+	const editor = 'classic' === action.editor ? 'gutenberg' : 'classic';
+	return bypassDataLayer( setSiteEditor( action.siteId, editor ) );
+};
```

Comment:
Not a blocker but just something I was wondering about.

This only works if `action.editor` is either `classic` or `gutenberg`.

But what if I try to set the editor to `foobar`?
My expectation would be to see it reverting to the previous value, but this instead would arbitrarily set it to `classic`.

It might makes sense now that we're in a transition phase, though, especially if our plans are to drop this opt-in altogether in the near-ish future. 🤷‍♂️